### PR TITLE
refactor: Remove stale references to flake8

### PR DESCRIPTION
**Pull request by romanc**
## Description

In PR https://github.com/GridTools/gt4py/pull/1466, `flake8`, `black`, and `isort` were replaced with the [ruff](https://astral.sh/ruff) linter and formatter. It seems there are a few leftovers of this transition. In this PR, I did a quick search for flake8 and deleted / replaced all instances found.

Removing `flake8` means rebuilding the requirements (to propagate the changes to the frozen `requirements-*` files), which updates a couple of dependencies.

We also decided to remove the `linter-*`  tasks of  `tox` since they have been replaced by `pre-commit`.

## Requirements

- [ ] All fixes and/or new features come with corresponding tests: N/A
- [ ] Important design decisions have been documented in the approriate ADR inside the [docs/development/ADRs/](docs/development/ADRs/Index.md) folder: N/A

If this PR contains code authored by new contributors please make sure:

- [x] The PR contains an updated version of the `AUTHORS.md` file adding the names of all the new contributors.

### CODE REVIEW

File: tox.ini
Comment (romanc) [inline tox.ini:106]: I think this is what triggered `flake8`  in https://github.com/GridTools/gt4py/pull/1624#issuecomment-2324035028.

Question here is whether we still need this category (at all) because, as far as I understand, the remaining `mypy`  is also called from `pre-commit`.

**Comment by egparedes** [comment]
Thanks for contributing with this PR to clean up GT4Py!
 
> @egparedes / @havogt: How can I propagate changes from requirements-dev.in to auto-generated files like requirements-dev.txt?

Just run: `tox r -e requirements-base`. More details here:  https://github.com/GridTools/gt4py/blob/main/docs/development/tools/requirements.md

File: tox.ini
Comment (egparedes) [reply tox.ini:106]: I agree, we should just completely delete the `linters` task in tox, since this has been fully replaced by pre-commit.

Comment (romanc) [reply tox.ini:106]: removed :broom:

**Comment by romanc** [comment]
@egparedes thanks for the quick reply and the pointers. I could run the `tox`  command to update the requirements and I checked the `mypy` version for consistency. This should be ready for review now.

/cc @FlorianDeconinck

**Comment by egparedes** [comment]
I hadn't realized that the CI checks were waiting for explicit approval of a member of the developers group before. I've trigger them now and waiting explicit

**Comment by egparedes** [comment]
cscs-ci run default

**Review by egparedes — APPROVED** [review]
Looks good to me.

**Comment by romanc** [comment]
Thanks for merging @egparedes. Happy to contribute!

**Pull request merged**
